### PR TITLE
AF-1646: if not offline, and a artifact has a bad description, we delete it and try resolve again.

### PR DESCRIPTION
**Pull request by porcelli**
There's also an improvement related to forced offline to not add maven central, what makes total sense.

### CODE REVIEW

**Comment by porcelli** [comment]
@jhrcek @ederign please take a look

**Comment by porcelli** [comment]
jenkins please retest this

**Comment by porcelli** [comment]
jenkins please retest this

**Comment by porcelli** [comment]
jenkins please retest this

**Comment by mbiarnes** [comment]
Jenkins please retest this

**Comment by porcelli** [comment]
jenkins execute full downstream build

File: kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenEmbedder.java [hunk 1]
```diff
@@ -292,18 +300,29 @@ public String getLocation() {
...
-            ProjectBuilder projectBuilder = componentProvider.lookup( ProjectBuilder.class );
-            // BZ-1007894: Check if added dependencies are resolvable.
-            ProjectBuildingResult result = projectBuilder.build( modelSource, getProjectBuildingRequest() );
-            if ( result != null && result.getDependencyResolutionResult() != null && !result.getDependencyResolutionResult().getCollectionErrors().isEmpty() ) {
-                // A dependency resolution error has been produced. It can contains some error. Throw the first one to the client, so the user will fix every one sequentially.
-                Exception depedencyResolutionException = result.getDependencyResolutionResult().getCollectionErrors().get( 0 );
-                if ( depedencyResolutionException != null ) {
-                    throw new MavenEmbedderException( depedencyResolutionException.getMessage(), depedencyResolutionException );
+            org.eclipse.aether.artifact.Artifact lastArtifact = null;
+            do {
```
Comment (jhrcek) [inline kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenEmbedder.java]: I tested this locally and the scenario described in AF-1646 is working fine with this fix.
I kinda don't like how convoluted this code is after the changes. infinite do-while loop? continue? mutable variables to control state of the looping? No comments explaining why we tryRemoveLocalArtifact?

I think this code could be made more readable

Comment (porcelli) [reply kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenEmbedder.java]: @jhrcek could you provide a suggestion? as for me, the code looks ok (of course I'm biased here).

File: kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenEmbedder.java [hunk 2]
```diff
@@ -292,18 +300,29 @@ public String getLocation() {
...
+                            continue;
+                        }
+                    }
+                    if (depedencyResolutionException != null) {
+                        throw new MavenEmbedderException(depedencyResolutionException.getMessage(), depedencyResolutionException);
+                    }
                 }
-            }
-            return result.getProject();
+                return (result == null || result.getProject() == null ) ? null : result.getProject();
```
Comment (jhrcek) [inline kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenEmbedder.java]: After staring at the code for some time I didn't find much to improve.
But the code could be simplified by removing some unnecessary nullchecks (up to you).
According to javadoc ProjectBuilder#build
`@return The result of the project building, never {@code null}.`
so the `result!=null` at 309 and `result == null` at 234 can be removed.

Comment (porcelli) [reply kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenEmbedder.java]: @jhrcek i don't have the confidence that ProjectBuilder#build won't ever be null.. would be better if it was wrapped in an Optional<T>, but would potentially be a bigger refactoring.. so I'd prefer to play safe here for now.

**Pull request merged**